asus/battery.nix

```diff
@@ -0,0 +1,21 @@
+{ config, pkgs, lib, ... }:
+let
+  p = pkgs.writeScriptBin "charge-upto" ''
+    echo ''${0:-100} > /sys/class/power_supply/BAT0/charge_control_end_threshold
+  '';
```

Comment:
That script does not provide a lot of value over `echo 100 > /sys/class/power_supply/BAT0/charge_control_end_threshold` and the script is not very discoverable.
Maybe leave it in your own configuration.

```suggestion
```

Reply:
Not that different from nvidia-offload
https://github.com/NixOS/nixos-hardware/blob/5a7e613703ea349fd46b3fa2f3dfe3bd5444d591/common/gpu/nvidia.nix#L6

Reply:
Sorry. I don't understand what you mean by this.

Reply:
`charge-upto` exposes a little known linux API via shell command, just like nvidia-offload.
Yes, both scrips are not much discoverable, but both provide value in just their name.

Reply:
nvidia-offload allows you to run things on your second GPU. You need both GPUs for that feature to work. This script is really generic and should work on almost every device with a recent Linux kernel and a battery.